packages/frontend/src/features/auth/components/ProtectedRoute.tsx

```diff
@@ -1,29 +1,19 @@
-import React, { Component } from "react";
+import React, { Component, ReactType } from "react";
 import { Route, Redirect } from "react-router-dom";
-import { loginSubmit } from "../action/authAction";
 import { Navigation } from "../../../config/routing/NavigationPath";
 
-export const ProtectedRoute = ({ component: Component, ...rest }: any) => {
-  const checkToken = localStorage.getItem("token");
-  return (
-    <Route
-      {...rest}
-      render={(props) => {
-        if (checkToken) {
-          return <Component {...props} />;
-        } else {
-          return (
-            <Redirect
-              to={{
-                pathname: Navigation.LOGIN,
-                state: {
-                  from: props.location,
-                },
-              }}
-            />
-          );
-        }
-      }}
-    />
-  );
+interface ProtectedProps {
+  component: ReactType;
+  path?: string;
+  exact?: boolean;
+}
+
+const ProtectedRoute = ({ component: Component, ...rest }: ProtectedProps) => {
+  const token = sessionStorage.getItem("token");
+  const handleTokenCheck = () => {
```

Comment:
To na zewnatrz albo do useCalback. Skoro handleTokenCheck to () =>JSX a props render spodziewa się ()=>JSX to poco opakowujesz to jeszcze w funkcje?